### PR TITLE
SCA card payment fix

### PR DESCRIPTION
**Pull request by AndriiMysko**
Hello @artursmirnov 
Just a quick fix for the SCA card payments, please take a look.

P.S. pollSubscription removal is intentional, it is no longer needed

### CODE REVIEW

File: app/components/billing/authorization.js [hunk 1]
```diff
@@ -46,7 +46,8 @@ export default Component.extend({
 
   retryAuthorization: task(function* () {
     if (this.requiresSourceAction && this.retryAuthorizationClientSecret) {
-      yield this.stripe.handleStripePayment.perform(this.retryAuthorizationClientSecret);
+      this.stripe.handleStripePayment.perform(this.retryAuthorizationClientSecret);
```
Comment (artursmirnov) [inline app/components/billing/authorization.js:49]: It feels to me that `yield` is still required here to wait for the payment task to perform, isn't it?

File: app/components/billing/payment.js [hunk 2]
```diff
@@ -133,7 +133,8 @@ export default Component.extend({
             token: token.id,
             lastDigits: token.card.last4
           });
-          yield subscription.save();
+          const { clientSecret } = yield subscription.save();
+          this.stripe.handleStripePayment.perform(clientSecret);
```
Comment (artursmirnov) [inline app/components/billing/payment.js:137]: Same here, `yield` seems to be required here.

File: app/models/v2-subscription.js [hunk 3]
```diff
@@ -22,7 +22,7 @@ export default Model.extend({
   organization: belongsTo('organization'),
   coupon: attr('string'),
   clientSecret: attr('string'),
-  paymentIntent: attr('string'),
+  paymentIntent: attr(),
```
Comment (artursmirnov) [inline app/models/v2-subscription.js:25]: Is this change intentional? What's the type of `paymentIntent`?

**Review by artursmirnov — CHANGES_REQUESTED** [review]

File: app/models/v2-subscription.js (hunk 3 above)
Comment (AndriiMysko) [reply app/models/v2-subscription.js:25]: This change is intentional, yes, it's an object

File: app/components/billing/payment.js (hunk 2 above)
Comment (AndriiMysko) [reply app/components/billing/payment.js:137]: When using `yield` here, subscription does not update after the payment is performed, while doing it this way does work.

Comment (artursmirnov) [reply app/components/billing/payment.js:137]: Would it make sense to investigate the root issue in this case? I'm afraid it could lead to some nasty bugs, because something seems to be wrong here.

Comment (sgolu6) [reply app/components/billing/payment.js:137]: @artursmirnov could you please clarify 'something seems to be wrong' ?
cc @shairyar

Comment (artursmirnov) [reply app/components/billing/payment.js:137]: @sgolu6 `handleStripePayment` is an asynchronous task that handles Stripe payment (sends request to Stripe and waits for response). This method is _aimed_ to run asynchronously and be awaited to get result. In the current implementation, the code just sends the payment request to Stripe and continues to subscription update without waiting for results from Stripe. If it doesn't work as intended, then there might be some kind of a hidden error that causes it. And as long as this is quite sensible part (handling the payment), I though, it wouldn't hurt to be careful with it.

**Review by shairyar — APPROVED** [review]

**Review by artursmirnov — APPROVED** [review]

File: app/components/billing/payment.js (hunk 2 above)
Comment (artursmirnov) [reply app/components/billing/payment.js:137]: As discussed internally with @shairyar, this is acceptable and we can leave it this way for now.

Comment (shairyar) [reply app/components/billing/payment.js:137]: Thank you @artursmirnov
